packages/flutter/lib/src/material/carousel.dart

```diff
@@ -1770,18 +1774,13 @@ class CarouselController extends ScrollController {
 
   /// The index of the leading item in the controlled carousel.
   ///
-  /// If there are no clients attached to this controller, returns [initialItem].
+  /// Returns [initialItem] if the carousel is not attached yet.
```

Comment:
I feel like it should assert in this case. If nothing has been laid out yet, we are trying to access a leadingItem that does not exist yet. That does not seem like a valid state.